x-pack/plugins/apm/public/context/MatchedRouteContext.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useMemo } from 'react';
+import { matchPath } from 'react-router-dom';
+import { routes } from '../components/app/Main/route_config';
+import { useLocation } from '../hooks/useLocation';
+
+export const MatchedRouteContext = React.createContext([] as Array<
+  typeof routes[0]
+>);
```

Comment:
Should be able to safely leverage the generic type variable for `React.createContext` rather than using the type assertion:
```typescript
export const MatchedRouteContext = React.createContext<typeof routes[0][]>([]);
```

Reply:
nice!